### PR TITLE
TS-4018: Use HPACK huffman encoding always

### PR DESCRIPTION
**Pull request by maskit**
https://issues.apache.org/jira/browse/TS-4018

### CODE REVIEW

**Comment by jpeach** [comment]
FYI it is better to squash the format commits rather than have them appear separately.

File: proxy/http2/HPACK.cc [hunk 1]
```diff
@@ -332,18 +332,41 @@ int64_t
 encode_string(uint8_t *buf_start, const uint8_t *buf_end, const char *value, size_t value_len)
 {
   uint8_t *p = buf_start;
+  bool use_huffman = true;
```
Comment (bryancall) [inline proxy/http2/HPACK.cc]: Why have a flat to use huffman if you are not going to change it to false?

File: proxy/http2/HPACK.cc [hunk 2]
```diff
@@ -332,18 +332,41 @@ int64_t
...
-  const int64_t len = encode_integer(p, buf_end, value_len, 7);
+  const int64_t len = encode_integer(p, buf_end, data_len, 7);
   if (len == -1)
     return -1;
+  if (use_huffman) {
+    *p |= 0x80;
+  }
   p += len;
-  if (buf_end < p || static_cast<size_t>(buf_end - p) < value_len)
+  if (buf_end < p || static_cast<size_t>(buf_end - p) < data_len)
```
Comment (bryancall) [inline proxy/http2/HPACK.cc]: I get this error when compiling:
HPACK.cc:359:55: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   if (buf_end < p || static_cast<size_t>(buf_end - p) < data_len)

File: proxy/http2/HPACK.cc (hunk 1 above)
Comment (maskit) [inline proxy/http2/HPACK.cc]: I left the flag for further use. There are some cases that plain text is shorter than huffman encoded data, and the codes for both cases would be used when we found a smart way to switch the encoding.

Comment (masaori335) [inline proxy/http2/HPACK.cc]: According to [Compression Ratio](https://github.com/http2jp/hpack-test-case/wiki/Compression-Ratio) in [hpack-test-case](https://github.com/http2jp/hpack-test-case), the best compression rate is `0.31` but compression rate of naive-huffman imprementation (I guess this is similar to this PR) is `0.85`. So we have to back to here and implement smart algorithm.

**Comment by maskit** [comment]
I fixed the compile error, and squashed the commits.

**Pull request closed without merging**